Document CI configuration and provide current status

My local build of [current main](https://github.com/cardano-foundation/cardano-wallet-agda/tree/f3479b501a2efe50bcf1ee0d09bc2d1325a982e7)failed to compile, giving the following error:

```
[42 of 71] Compiling Cardano.Wallet.Read.Chain ( haskell/Cardano/Wallet/Read/Chain.hs, /Users/arnaud/projects/cardano-wallet-agda/dist-newstyle/build/aarch64-osx/ghc-9.10.1/cardano-wallet-read-0.2024.8.27/build/Cardano/Wallet/Read/Chain.o, /Users/arnaud/projects/cardano-wallet-agda/dist-newstyle/build/aarch64-osx/ghc-9.10.1/cardano-wallet-read-0.2024.8.27/build/Cardano/Wallet/Read/Chain.dyn_o )
haskell/Cardano/Wallet/Read/Chain.hs:22:7: error: [GHC-97219]
    Duplicate record field ‘headerHash’ in export list:
       ‘ChainTip(..)’ exports the field ‘headerHash’
       belonging to the constructor ‘BlockTip’
         imported from ‘Cardano.Wallet.Read.Chain.Point’ at haskell/Cardano/Wallet/Read/Chain.hs:54:7-19
         (and originally defined
            at haskell/Cardano/Wallet/Read/Chain/Point.hs:117:11-20)
       ‘ChainPoint(..)’ exports the field ‘headerHash’
       belonging to the constructor ‘BlockPoint’
         imported from ‘Cardano.Wallet.Read.Chain.Point’ at haskell/Cardano/Wallet/Read/Chain.hs:53:7-21
         (and originally defined
            at haskell/Cardano/Wallet/Read/Chain/Point.hs:77:11-20)
    Suggested fix: Perhaps you intended to use DuplicateRecordFields
   |
22 |     , ChainTip (..)
   |       ^^^^^^^^^^^^^

haskell/Cardano/Wallet/Read/Chain.hs:22:7: error: [GHC-97219]
    Duplicate record field ‘slotNo’ in export list:
       ‘ChainTip(..)’ exports the field ‘slotNo’
       belonging to the constructor ‘BlockTip’
         imported from ‘Cardano.Wallet.Read.Chain.Point’ at haskell/Cardano/Wallet/Read/Chain.hs:54:7-19
         (and originally defined
            at haskell/Cardano/Wallet/Read/Chain/Point.hs:116:11-16)
       ‘ChainPoint(..)’ exports the field ‘slotNo’
       belonging to the constructor ‘BlockPoint’
         imported from ‘Cardano.Wallet.Read.Chain.Point’ at haskell/Cardano/Wallet/Read/Chain.hs:53:7-21
         (and originally defined
            at haskell/Cardano/Wallet/Read/Chain/Point.hs:76:11-16)
    Suggested fix: Perhaps you intended to use DuplicateRecordFields
   |
22 |     , ChainTip (..)
   |       ^^^^^^^^^^^^^

```

In retrospect it seems obvious this is caused by a discrepancy in GHC versions (I am not using nix) as the existing [CI pipeline](https://buildkite.com/cardano-foundation/cardano-wallet-agda/builds/389) is green. However I overlooked this information as the only indication of CI results is in the commit line, and not at the top of the README page as is customary in a lot of OSS projects. 

I think it would be good to have a link in the `README.md`, e.g. like traditional badges, and perhaps also document how CI is done in the project. 

**Note 1**: an additional issue is that CI is not accessible to people outside of CF organisation, perhaps we should consider a different option?

**Note 2**: perhaps we want to make the use of nix mandatory to build this project?

--- a/.github/PULL_REQUEST_TEMPLATE.md
+++ b/.github/PULL_REQUEST_TEMPLATE.md
@@ -14,7 +14,3 @@ Describe in more detail, in a few bullet points, of how this PR accomplishes its
 ### Comments
 
 <!-- Additional comments on the why of this pull request, links, or screenshots to attach, if any. -->
-
-### Issue Number
-
-<!-- Reference the Jira/GitHub issue that this PR relates to, and which requirements it tackles. -->

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,3 +1,23 @@
 If you find a bug or you'd like to propose a feature, please feel free to raise an issue on our [issue tracker](https://github.com/cardano-foundation/cardano-wallet-agda/issues).
 
 Please note we have a [code of conduct](CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
+
+# Development
+
+## Branches
+
+We use [trunk-based development][trunk]: We merge features and improvements into the `main` branch frequently. If a large feature takes more than one pull request to develop, and is therefore incomplete, we hide it behind a feature flag. The `main` branch should be kept in a state where all checks pass, so that we can release it at any time.
+
+  [trunk]: https://martinfowler.com/articles/branching-patterns.html#Trunk-basedDevelopment
+
+## Continuous Integration (CI)
+
+We use [continuous integration][ci]: Checking proofs, building artifacts, and testing artifacts is done automatically before changes are admitted to the `main` branch.
+
+We use [Buildkite][] for automation.
+Our Buildkite pipeline is defined in [.buildkite/pipeline.yml](.buildkite/pipeline.yml). The build environment is provisioned using the [Nix package manager][nix]. The relevant build steps are grouped in the [`justfile`][just] under the `ci-` prefix.
+
+  [ci]: https://www.goodreads.com/book/show/17255186-the-phoenix-project
+  [nix]: https://nixos.org/download/
+  [buildkite]: https://buildkite.com/cardano-foundation/cardano-wallet-agda/
+  [just]: https://github.com/casey/just

--- a/README.md
+++ b/README.md
@@ -29,7 +29,8 @@ Each of the packages in the `lib/` directory may contain the following subdirect
 
 Prerequisites:
 
-* [Nix package manager][nix].
+* [Nix package manager][nix].  
+  As some dependencies, such as [agda2hs][], are still under development, we use [nix][] to ensure a working build environment.
 
 How to build:
 

--- a/README.md
+++ b/README.md
@@ -1,6 +1,8 @@
 
 🚧 WORK IN PROGRESS 🚧
 
+[![Build status](https://badge.buildkite.com/307068367b3a4746737164e239dca22dd6b5288eb54c71ee0c.svg?branch=main)](https://buildkite.com/cardano-foundation/cardano-wallet-agda)
+
 ## Overview
 
 This repository contains wallet implementations for the Cardano blockchain that are formally verified to some degree. The implementations are [Haskell][] packages, but generated using [agda2hs][], with proofs in [Agda][].

